server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStats.java

```diff
@@ -24,70 +27,133 @@
  */
 public class RemoteStoreStats implements Writeable, ToXContentFragment {
 
-    private final RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats;
+    private final RemoteRefreshSegmentTracker.Stats remoteSegmentShardStats;
 
-    public RemoteStoreStats(RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats) {
-        this.remoteSegmentUploadShardStats = remoteSegmentUploadShardStats;
+    private final ShardRouting currentRouting;
+
+    public RemoteStoreStats(RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats, ShardRouting currentRouting) {
+        this.remoteSegmentShardStats = remoteSegmentUploadShardStats;
+        this.currentRouting = currentRouting;
     }
 
     public RemoteStoreStats(StreamInput in) throws IOException {
-        remoteSegmentUploadShardStats = in.readOptionalWriteable(RemoteRefreshSegmentTracker.Stats::new);
+        this.remoteSegmentShardStats = in.readOptionalWriteable(RemoteRefreshSegmentTracker.Stats::new);
```

Comment:
Why does `remoteSegmentShardStats` need to be read optionally?

Reply:
I think there is no other way to read

Reply:
Yes. All the other APIs are also using the same `readOptionalWriteable` method. Seems like we do not have a direct `readWriteable` alternative.